lib/src/main/java/tech/relaycorp/awaladroid/AndroidPrivateKeyStore.kt

```diff
@@ -6,29 +6,41 @@ import androidx.security.crypto.MasterKey
 import tech.relaycorp.awala.keystores.file.FileKeystoreRoot
 import tech.relaycorp.awala.keystores.file.FilePrivateKeyStore
 import java.io.File
+import javax.crypto.AEADBadTagException
 
 internal class AndroidPrivateKeyStore(
     root: FileKeystoreRoot,
     private val context: Context,
 ) : FilePrivateKeyStore(root) {
+
+    @Throws(EncryptionInitializationException::class)
     override fun makeEncryptedInputStream(file: File) = buildEncryptedFile(file).openFileInput()
 
+    @Throws(EncryptionInitializationException::class)
     override fun makeEncryptedOutputStream(file: File) = buildEncryptedFile(file).openFileOutput()
 
+    @Throws(EncryptionInitializationException::class)
     private fun buildEncryptedFile(file: File) =
-        EncryptedFile.Builder(
-            context,
-            file,
-            masterKey,
-            EncryptedFile.FileEncryptionScheme.AES256_GCM_HKDF_4KB,
-        ).build()
+        try {
+            EncryptedFile.Builder(
+                context,
+                file,
+                masterKey,
+                EncryptedFile.FileEncryptionScheme.AES256_GCM_HKDF_4KB,
+            ).build()
+        } catch (e: AEADBadTagException) {
+            // Known issue: https://issuetracker.google.com/issues/164901843
+            throw EncryptionInitializationException(e)
+        }
 
     private val masterKey by lazy {
         MasterKey.Builder(context, MASTER_KEY_ALIAS)
             .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
             .build()
     }
 
+    class EncryptionInitializationException(cause: Throwable) : Exception(cause)
```

Comment:
Apps won't be able to catch this exception: It's inside an `internal` class. Can you please promote it to a `public`, top-level class?